### ISSUE TITLE
popup width

### ISSUE THREAD

**Issue by cotcomsol** [TerminalFi/LSP-copilot#117]
Is there a way to change the popup width?  So far I have been unable to find a setting.  On my system it always shows up too narrow and the completions are cut off.  My sublime window in this screenshot is over 3000 pixels wide so plenty of space for it to display the full completion if it just made the popup wider.
![Screenshot_2023-08-07_09-06-29](https://github.com/TerminalFi/LSP-copilot/assets/414939/54ce4836-87f9-4685-a0ac-0199f54a5294)

**Comment by TerminalFi** [comment]
The short answer, not currently. The long answer is it impossible believe to make this user a user configuration. But not sure if it is really a high priority. But I do see how this can make it not as usable.

**Comment by timfjord** [comment]
Just a note, you could use
```json
"settings": {
  "completion_style": "popup"
}
```
to have a VSCode-like completion style.
But please note it is a bit buggy

**Comment by TerminalFi** [comment]
After months, releasing this within the week

**Comment by WangWei90** [comment]
> Just a note, you could use
> 
> ```json
> "settings": {
>   "completion_style": "phantom"
> }
> ```
> 
> to have a VSCode-like completion style. But please note it is a bit buggy

Can the "phantom" be setup with a different background color, can the "phantom" use the upright style?

**Comment by timfjord** [comment]
@WangWei90 I don't think we set the background colour explicitly, but since the package follows [the best practices](https://www.sublimetext.com/docs/minihtml.html#best-practices) it should be possible to customize it if needed.

I'm not sure what the upright style means though. But if it is about styling, then it should be possible to customize it too.

**Comment by WangWei90** [comment]
Thank you~ @timfjord 

I will try to test this later.

**Comment by TerminalFi** [comment]
@cotcomsol Is this still desired?

**Comment by cotcomsol** [comment]
It would definitely be helpful for me.  I find that most of the time the completion is cut off and so mostly useless.

**Comment by jfcherng** [comment]
or maybe we show it when the time elapses a thrshold

**Comment by TerminalFi** [comment]
> or maybe we show it when the time elapses a thrshold

Wrong thread :)

**Comment by jfcherng** [comment]
Hey @TerminalFi are you going to review https://github.com/TerminalFi/LSP-copilot/pull/178 ? I see you close this issue without merging that PR.

**Comment by jfcherng** [comment]
Hmm... I guess #178 won't help. The popup width is provided as an argument to a ST plugin API.

**Comment by TerminalFi** [comment]
It helps, I'll post example. I've tested this already.

**Comment by jfcherng** [comment]
> It helps, I'll post example. I've tested this already.

Oops, I thought it's because of https://github.com/TerminalFi/LSP-copilot/blob/fa70eafc1916737bf623480497ef34dcae61d1e1/plugin/ui/completion.py#L244

**Comment by TerminalFi** [comment]
> > It helps, I'll post example. I've tested this already.
> 
> 
> 
> Oops, I thought it's because of https://github.com/TerminalFi/LSP-copilot/blob/fa70eafc1916737bf623480497ef34dcae61d1e1/plugin/ui/completion.py#L244

That's just passed as CSS I believe. But either way, I'll post example

**Comment by TerminalFi** [comment]
Actually, I had local changes!

We do need to expose max_width and max_height as well.

**Comment by TerminalFi** [comment]
@jfcherng we can either set higher defaults or create new settings

```

		"completion_popup_max_height": "240",
		"completion_popup_max_width": "640",
```

**Comment by jfcherng** [comment]
Will it work if we hard-code it like `max_width=9999` and limit it with CSS?

**Comment by jfcherng** [comment]
> @jfcherng we can either set higher defaults or create new settings
> 
> ```
> 
> 		"completion_popup_max_height": "240",
> 		"completion_popup_max_width": "640",
> ```

Then yes, I think we have no choice.
